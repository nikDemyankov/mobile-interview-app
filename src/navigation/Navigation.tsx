import React, { memo } from 'react';
import { createStackNavigator } from '@react-navigation/stack';

import WelcomeScreen from '../screen/welcome';
import InstrumentListScreen from '../screen/instrumentList';

const Stack = createStackNavigator();

const Navigation: React.FC = () => {
  return (
    <Stack.Navigator>
      <Stack.Screen
        name="Welcome"
        component={WelcomeScreen}
        options={{
          headerShown: false,
        }}
      />
      <Stack.Screen name="InstrumentList" component={InstrumentListScreen} />
    </Stack.Navigator>
  );
};

export default memo(Navigation);
