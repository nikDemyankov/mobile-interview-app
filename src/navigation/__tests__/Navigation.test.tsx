import React from 'react';
import { create } from 'react-test-renderer';
import Navigation from '../Navigation';

jest.mock('@react-navigation/stack', () => ({
  createStackNavigator: () => ({
    Navigator: 'Stack.Navigator',
    Screen: 'Stack.Screen',
  }),
}));
jest.mock('../../screen/welcome', () => 'WelcomeScreen');
jest.mock('../../screen/instrumentList', () => 'InstrumentListScreen');

it('renders correct stack', () => {
  const component = create(<Navigation />);
  expect(component).toMatchSnapshot();
});
