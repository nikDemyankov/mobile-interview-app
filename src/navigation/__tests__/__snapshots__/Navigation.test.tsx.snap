// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct stack 1`] = `
<Stack.Navigator>
  <Stack.Screen
    component="WelcomeScreen"
    name="Welcome"
    options={
      Object {
        "headerShown": false,
      }
    }
  />
  <Stack.Screen
    component="InstrumentListScreen"
    name="InstrumentList"
  />
</Stack.Navigator>
`;
